### ISSUE TITLE
pie takes negative values 

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#15923]
### Bug report

**Bug summary**
`Axes.pie` accepts negative values which leads to incorrect normalization and 
**Code for reproduction**

```python
# Paste your code here
#
#
fig, ax = plt.subplots()

ax.pie([5, 5, -3], explode=[0, .1, .2])


```

**Actual outcome**

![womp_womp](https://user-images.githubusercontent.com/199813/70752648-0a1bfa80-1d01-11ea-8ea0-4104c5d7a2f2.png)

**Expected outcome**

This should raise an exception 

**Matplotlib version**

  * Operating system:
  * Matplotlib version:  '3.1.1.post2607+g56905d247'
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt4Agg
  * Python version: '3.7.4'

**Comment by Inception95** [comment]
Hi, I am new here. Can I help with this issue if it is not been solved yet? Thanks!

**Comment by jklymak** [comment]
Yes go for it.  See the somewhat daunting: https://matplotlib.org/3.1.1/devel/contributing.html

**Comment by timhoffm** [comment]
Closed by #15935.
